consent/handler.go

```diff
@@ -133,6 +134,47 @@ func (h *Handler) DeleteUserClientConsentSession(w http.ResponseWriter, r *http.
 	w.WriteHeader(http.StatusNoContent)
 }
 
+// swagger:route GET /oauth2/auth/sessions/consent/{user} oAuth2 listUserClientConsentSessions
+//
+// List all consent sessions of a user
+//
+// This endpoint lists all user's granted consent sessions, including client and granted scope
+//
+//     Consumes:
+//     - application/json
+//
+//     Produces:
+//     - application/json
+//
+//     Schemes: http, https
+//
+//     Responses:
+//       200: handledConsentRequestList
+//       401: genericError
+//       403: genericError
+//       500: genericError
+
+func (h *Handler) GetConsentSessions(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	user := ps.ByName("user")
+	if user == "" {
+		h.H.WriteError(w, r, errors.WithStack(fosite.ErrInvalidRequest.WithDebug("Parameter user is not defined")))
+		return
+	}
+
+	sessions, err := h.M.FindPreviouslyGrantedConsentRequestsByUser(user)
```

Comment:
I really think we need pagination here, because this can be a lot! Examples can be found [here](https://github.com/ory/hydra/blob/master/client/handler.go#L192-L193), [here](https://github.com/ory/hydra/blob/master/client/manager_sql.go#L377-L381) and [here](https://github.com/ory/hydra/blob/master/client/manager_memory.go#L151-L154).

Reply:
Thanks, I adapted it. However, I was a bit unsure in `consent/manager_memory.go ` as `FindPreviouslyGrantedConsentRequests` depends now on the newly created `FindPreviouslyGrantedConsentRequestsByUser` (which requires a limit and offset now) and does not support limit and offset. As you'll see, I've used a little workaround that I'm personally not 100% satisfied with.

Reply:
I see, I don't think this solution will work with SQL as they don't support negative offset (I think).

Reply:
Oh sorry, I didn't see that you duplicated the other method. Then it will obviously work. I agree that it's not the cleanest solution but it does get the job done for now. So from my side, this is fine.